src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/visit/CancelVisitTest.kt

```diff
@@ -280,4 +290,197 @@ class CancelVisitTest(@Autowired private val objectMapper: ObjectMapper) : Integ
     verify(telemetryClient, times(1)).trackEvent(eq("visit-bad-request-error"), any(), isNull())
     verify(telemetryClient, times(0)).trackEvent(eq("visit.cancelled-domain-event"), any(), isNull())
   }
+
+  /**
+   * the check for cancellations does not calculate time - only dates.
+   */
+  @Test
+  fun `cancel expired visit on same day as allowed day does not return error`() {
+    val outcomeDto = OutcomeDto(
+      OutcomeStatus.CANCELLATION,
+      "No longer joining."
+    )
+    // Given
+    val visitStart = LocalDateTime.now().minusDays(visitCancellationDayLimit).truncatedTo(ChronoUnit.DAYS).withHour(1)
+    val expiredVisit = visitEntityHelper.create(visitStatus = BOOKED, visitStart = visitStart)
+
+    // When
+    val responseSpec = callCancelVisit(webTestClient, setAuthorisation(roles = listOf("ROLE_VISIT_SCHEDULER")), expiredVisit.reference, outcomeDto)
+
+    // Then
+    responseSpec.expectStatus().isOk
+      .expectBody()
+      .jsonPath("$.visitStatus").isEqualTo(VisitStatus.CANCELLED.name)
+      .jsonPath("$.outcomeStatus").isEqualTo(OutcomeStatus.CANCELLATION.name)
```

Comment:
only have to check for cancelled status...

as other test cover out come!
"cancel visit by reference -  with outcome and outcome text"

Or you could have a common method something like 
assertVisitCancellation() and in that you could check all asserts e.g.

 status
 outcome 
 events

and reuse across test class

Reply:
Updated based on the above suggestions. thanks.